### PR TITLE
Uniformize Warnings & 100 Character Line Width 

### PR DESCRIPTION
**Pull request by InfiniteEchoes**
This PR is for #477 and #497 , since the change to fix them are pretty small.
- Made the `note` for `emit_warning_with_note` optional
- Regarded other minor changes
- Adjusted line width to 100

### CODE REVIEW

File: lib/src/env.rs [hunk 1]
```diff
@@ -12,11 +12,14 @@ pub(crate) fn emit_warning_with_note(
     env: &Env,
     span: &rustc_span::Span,
     warning_msg: &str,
-    note_msg: &str,
+    note_msg: &Option<&str>,
```
Comment (clarus) [inline lib/src/env.rs:15]:
```suggestion
    note_msg: Option<&str>,
```

File: lib/src/env.rs [hunk 2]
```diff
@@ -12,11 +12,14 @@ pub(crate) fn emit_warning_with_note(
...
 ) {
-    env.tcx
+    let mut warn = env
```
Comment (clarus) [inline lib/src/env.rs:17]:
```suggestion
    let warn = env
```

Comment (InfiniteEchoes) [reply lib/src/env.rs:17]: If I remove the `mut`, the compiler would report that it cannot be borrowed as mutable in the match branches. Is there a way to solve this issue or I can just keep the `mut`?

Comment (clarus) [reply lib/src/env.rs:17]: OK yes I did not know, so we keep the `mut` then!

**Review by clarus — APPROVED** [review]

**Comment by clarus** [comment]
OK, we can merge once the CI is green!

**Pull request merged**